File: GuardLAN.UI/src/app/features/integrations/models/integration-health.ts
export type IntegrationKind = 'Dns' | 'Zeek' | 'Suricata';

export type IntegrationHealthStatus = 'Healthy' | 'Warning' | 'Unavailable' | 'Disabled';

export interface IntegrationHealthOverviewDto {
  readonly summary: IntegrationHealthSummaryDto;
  readonly sources: readonly IntegrationHealthDto[];
}

export interface IntegrationHealthSummaryDto {
  readonly totalSources: number;
  readonly healthySources: number;
  readonly warningSources: number;
  readonly unavailableSources: number;
  readonly disabledSources: number;
  readonly lastCheckedUtc: string | null;
}

export interface IntegrationHealthDto {
  readonly id: string;
  readonly source: string;
  readonly kind: IntegrationKind;
  readonly status: IntegrationHealthStatus;
  readonly sourceEnabled: boolean;
  readonly sourceAvailable: boolean;
  readonly lastCheckedUtc: string;
  readonly lastSuccessUtc: string | null;
  readonly lastFailureUtc: string | null;
  readonly recordsRead: number;
  readonly recordsImported: number;
  readonly recordsRejected: number;
  readonly message: string;
}

File: GuardLAN.API/src/GuardLan.Domain/Entities/IntegrationHealth.cs
using GuardLan.Domain.Enums;

namespace GuardLan.Domain.Entities;

public class IntegrationHealth
{
    public Guid Id { get; set; }

    public string Source { get; set; } = null!;

    public IntegrationKind Kind { get; set; }

    public IntegrationHealthStatus Status { get; set; }

    public bool SourceEnabled { get; set; }

    public bool SourceAvailable { get; set; }

    public DateTime LastCheckedUtc { get; set; }

    public DateTime? LastSuccessUtc { get; set; }

    public DateTime? LastFailureUtc { get; set; }

    public int RecordsRead { get; set; }

    public int RecordsImported { get; set; }

    public int RecordsRejected { get; set; }

    public string Message { get; set; } = null!;
}
